File: tests/function/function.cpp
#include <iostream>
#include <string>

int add(int a, int b) {
    return (a + b);
}

void print_sum(int a, int b) {
    s = add(a, b);
    std::cout << "Sum = " << " " << s << std::endl;
}

int main() {
    int x, y, s;
    
    x = 10;
    y = add(x, 5);
    std::cout << "Result of function add: " << " " << y << std::endl;
    print_sum(3, 4);
    return 0;
}

File: tests/function/function.pas
program test;
var
    x, y, s: integer;

function add(a: integer; b: integer): integer;
begin
    add := a + b;
end;

procedure print_sum(a: integer; b: integer);
begin
    s := add(a, b);
    writeln('Sum = ', s);
end;

begin
    x := 10;
    y := add(x, 5);
    writeln('Result of function add: ', y);

    print_sum(3, 4);
end.